packages/api/lib/types.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * @typedef {Object} NormalizedRecReportParams
+ * @property {string[]} [collectionIds] - An optional array of collection IDs.
+ * @property {string[]} [granuleIds] - An optional array of granule IDs.
+ * @property {string[]} [providers] - An optional array of provider names.
+ * @property {string} [startTimestamp] - An optional start timestamp for the report.
+ * @property {string} [endTimestamp] - An optional end timestamp for the report.
+ * @property {string} [reportType] - An optional type of the report.
+ * @property {string} [location]
+ * @property {string} stackName
+ * @property {string} systemBucket
+ * @property {string} [status] - Optional granule status filter for report
+ */
+
+/**
```

Comment:
is there no way to simplify this? (unless I'm missing something) this "enhancedParams" is used only to define EnhancedNormalizedRecReportParams. is there a reason not to define EnhancedNormalziedRecReportParams as a child of NormalizedRecReportParams, and maybe NormalizedRecReportParams as a child of recReportParams? this feels like a more logical and understandable structure to me

Reply:
@etcart  100% agree, but...the union type worked with tscheck/jsdoc annotations whereas using typescript inheritance did not and sent me down more of a rabbit hole than that refactor felt worth at the time.   I'd love an example of this working, and agree enough I'll spend a bit more time on it.

Reply:
@etcart  re-reading this, I think I grok what you were getting at - the problem is this sort of thing: 

```
/**
 * @typedef {NormalizedRecReportParams} EnhancedNormalizedRecReportParams
```

Winds up not properly representing the combined type.   I'm uncertain if there's a better way to handle this.

Reply:
(talked about this in a huddle, closing with @etcart 's approval)